crates/transaction-emitter-lib/src/transaction_generator/nft_mint_and_transfer.rs

```diff
@@ -8,48 +9,91 @@ use aptos_sdk::{
 };
 
 use crate::emitter::{account_minter::create_and_fund_account_request, RETRY_POLICY};
+use aptos_infallible::Mutex;
 use aptos_logger::{info, warn};
+use aptos_sdk::types::account_address::AccountAddress;
+use async_trait::async_trait;
 use rand::rngs::StdRng;
+use rand::thread_rng;
 use std::sync::Arc;
 
-pub struct NFTMint {
+const INITIAL_NFT_BALANCE: u64 = 50_000;
+
+pub struct NFTMintAndTransfer {
     txn_factory: TransactionFactory,
-    creator_account: Arc<LocalAccount>,
+    creator_account: Arc<Mutex<LocalAccount>>,
+    faucet_account: LocalAccount,
     collection_name: Vec<u8>,
     token_name: Vec<u8>,
 }
 
-impl NFTMint {
+impl NFTMintAndTransfer {
     pub fn new(
         txn_factory: TransactionFactory,
-        creator_account: Arc<LocalAccount>,
+        creator_account: Arc<Mutex<LocalAccount>>,
         collection_name: Vec<u8>,
         token_name: Vec<u8>,
     ) -> Self {
+        let faucet_account = LocalAccount::generate(&mut thread_rng());
         Self {
             txn_factory,
+            faucet_account,
             creator_account,
             collection_name,
             token_name,
         }
     }
 }
 
-impl TransactionGenerator for NFTMint {
+#[async_trait]
+impl TransactionGenerator for NFTMintAndTransfer {
+    async fn setup(&mut self, rest_client: &RestClient, accounts: &[LocalAccount]) {
+        let creator_address = self.creator_account.lock().address();
+        let fund_faucet_account_txn = create_nft_transfer_request(
+            &mut self.creator_account.lock(),
+            &self.faucet_account,
+            creator_address,
+            &self.collection_name,
+            &self.token_name,
+            &self.txn_factory,
+            1_000_000_000,
+        );
+        submit_retry_and_wait(rest_client, &fund_faucet_account_txn).await;
+
+        for account in accounts {
+            let initial_nft_funding_txn = create_nft_transfer_request(
+                &mut self.faucet_account,
+                account,
+                creator_address,
+                &self.collection_name,
+                &self.token_name,
+                &self.txn_factory,
+                INITIAL_NFT_BALANCE,
+            );
+            submit_retry_and_wait(rest_client, &initial_nft_funding_txn).await;
+        }
+    }
+
     fn generate_transactions(
         &mut self,
         accounts: Vec<&mut LocalAccount>,
         transactions_per_account: usize,
     ) -> Vec<SignedTransaction> {
         let mut requests = Vec::with_capacity(accounts.len() * transactions_per_account);
         for account in accounts {
-            for _ in 0..transactions_per_account {
+            for i in 0..transactions_per_account {
```

Comment:
wouldn't this just work for the first time? 

should we every other call fund this account and leave it with INITIAL_NFT_BALANCE balance at the end?

Reply:
did the change.